src/utils/address.ts

```diff
@@ -0,0 +1,17 @@
+import { ethers } from "ethers";
+
+export const ethereumAddressRegex = "^0x[a-fA-F0-9]{40}$";
+export const ensRegex = "[a-z]+.eth$";
+
+export function isValidAddress(address: string): boolean {
+  if (typeof address !== "string") {
+    return false;
+  }
+  return address.match(ensRegex)
+    ? ethers.utils.isValidName(address)
+    : ethers.utils.isAddress(address);
```

Comment:
nice touch here to support ens. have never used it before, will this affect the claims logic if ens is used?

Reply:
I _think_ that if handled an ens domain ethers will resolve that to a valid address by itself. But let me actually test this and get back to you